### PR TITLE
[TECHNICAL-SUPPORT] LPS-90188

### PR DESCRIPTION
**Pull request by JosemariaMunoz**
Hi @stian-sigvartsen! 

This [LPS-90188](https://issues.liferay.com/browse/LPS-90188) is based on [PTR-591](https://issues.liferay.com/browse/PTR-591), please let me know any information that you need or any doubt that you could have.

Thanks!

### CODE REVIEW

**Comment by JosemariaMunoz** [comment]
ci:test

**Comment by JosemariaMunoz** [comment]
ci:retest

**Comment by lipusz** [comment]
ci:test:sf

**Comment by lipusz** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
Hey @JosemariaMunoz . Just starting to review this, sorry for the delay.

**Comment by JosemariaMunoz** [comment]
No problem @stian-sigvartsen, thank you very much!

File: modules/apps/portal-security/portal-security-ldap-impl/src/main/java/com/liferay/portal/security/ldap/internal/authenticator/LDAPAuth.java [hunk 1]
```diff
@@ -516,10 +527,24 @@ protected int authenticate(
...
+			else if (result == DNE) {
+				anyDne = true;
+			}
 		}
 
-		return authenticateRequired(
+		int authResult = authenticateRequired(
 			companyId, userId, emailAddress, screenName, true, FAILURE);
+
+		if ((authResult != SUCCESS) || (anyDne && !anyFailure)) {
```
Comment (stian-sigvartsen) [inline modules/apps/portal-security/portal-security-ldap-impl/src/main/java/com/liferay/portal/security/ldap/internal/authenticator/LDAPAuth.java]: Hey @JosemariaMunoz. I am curious why you are checking `anyDne` here? I am concerned that the resulting behavior becomes a bit sensitive to the user being remove from only one of a number of connected LDAP servers.
In such scenarios the behavior would revert to the current behavior.

**Comment by stian-sigvartsen** [comment]
Thanks! 👍 
Please consider https://github.com/JosemariaMunoz/liferay-portal/pull/46

File: modules/apps/portal-security/portal-security-ldap-impl/src/main/java/com/liferay/portal/security/ldap/internal/authenticator/LDAPAuth.java (hunk 1 above)
Comment (JosemariaMunoz) [reply modules/apps/portal-security/portal-security-ldap-impl/src/main/java/com/liferay/portal/security/ldap/internal/authenticator/LDAPAuth.java]: Thanks @stian-sigvartsen for your time, I answered you in the solution that you sent me: https://github.com/JosemariaMunoz/liferay-portal/pull/46#issuecomment-464007063
Please tell me if you need more information or anything you need!

**Comment by lipusz** [comment]
ci:close

**Pull request closed without merging**